### PR TITLE
[sdl] Update to version 2.0.16. JB#54674

### PR DESCRIPTION
**Pull request by mlehtima**
Use cmake for build. Remove obsolete patches.
Bring back wl-shell support because we don't support xdg-shell.

### CODE REVIEW

**Review by mkosola — APPROVED** [review]

File: rpm/sdl2.spec [hunk 1]
```diff
@@ -42,11 +41,20 @@ to develop SDL applications.
...
-%make_build
+mkdir -p build
+pushd build
+%cmake .. \
+  -DLIB_SUFFIX="" \
+  -DPULSEAUDIO=ON \
+  -DVIDEO_WAYLAND=ON \
+  -DVIDEO_X11=OFF \
+  -DWAYLAND_LIBDECOR=OFF
+popd
```
Comment (okodron) [inline rpm/sdl2.spec:52]: You can build it in current source directory:
```suggestion
%cmake \
  -DLIB_SUFFIX="" \
  -DPULSEAUDIO=ON \
  -DVIDEO_WAYLAND=ON \
  -DVIDEO_X11=OFF \
  -DWAYLAND_LIBDECOR=OFF
```

**Review by okodron — COMMENTED** [review]
Not related directly to this change but can you add `%license LICENSE.txt` as well?

File: rpm/sdl2.spec (hunk 1 above)
Comment (mlehtima) [reply rpm/sdl2.spec:52]: When trying to build in source directory the build fails:
```
CMake Error at CMakeLists.txt:2 (message):
  Prevented in-tree build.  Please create a build directory outside of the
  SDL source code and run "cmake -S some_path -B ."
  from there
```

Comment (okodron) [reply rpm/sdl2.spec:52]: Ok, got it.

**Review by okodron — APPROVED** [review]

**Review by mkosola — APPROVED** [review]

**Pull request merged**